### PR TITLE
Issue LIBCLOUD-335: Add IAM Profile on node creation for ec2 based providers

### PR DESCRIPTION
**Pull request by johnnoone**
Added ex_iamprofile parameter on node creation.
Should resolve https://issues.apache.org/jira/browse/LIBCLOUD-335

### CODE REVIEW

**Comment by Kami** [comment]
Thanks.

The pull request looks good, but it's missing tests. Once you add tests, ping me or any other developer\ here and we'll review it again.

If everything looks OK you will then be able to create a patch and a JIRA ticket and attach it there.

As far as the LIBCLOUD-335 ticket goes - this pull request shouldn't affect it in any way. This pull request changes EC2 compute driver which is unrelated to the storage driver mentioned in the ticket.

File: libcloud/compute/drivers/ec2.py [hunk 1]
```diff
@@ -1412,6 +1417,16 @@ def create_node(self, **kwargs):
                 for k, v in mapping.items():
                     params['BlockDeviceMapping.%d.%s' % (idx, k)] = str(v)
 
+        if 'ex_iamprofile' in kwargs:
+            try:
+                if kwargs['ex_iamprofile'].startswith('arn:aws:iam:'):
+                    params['IamInstanceProfile.Arn'] = kwargs['ex_iamprofile']
+                else:
+                    params['IamInstanceProfile.Name'] = kwargs['ex_iamprofile']
+            except AttributeError as exception:
```
Comment (Kami) [inline libcloud/compute/drivers/ec2.py]: `except as` syntax won't work in all the supported versions of Libcloud.

In fact, I would prefer for this to be changed to a more specific `isinstance(kwargs['ex_iamprofile'], basestring)` and throw if it's not.

**Comment by Kami** [comment]
@johnnoone Thanks, the update looks good.

Can you please go ahead, create a JIRA ticket (issues.apache.org/jira/browse/LIBCLOUD) and attach a patch there? I'll then go ahead and merge it into trunk.

**Comment by johnnoone** [comment]
here we are : https://issues.apache.org/jira/browse/LIBCLOUD-403
thanks.

**Comment by Kami** [comment]
Merged into trunk, you can close this pull request now.

**Comment by johnnoone** [comment]
thanks

**Comment by Kami** [comment]
You're welcome!

**Pull request closed without merging**